src/routes/index.js

```diff
@@ -12,7 +13,16 @@ const AppRouter = () => (
     <Route exact path="/dashboard" component={DashboardPage} />
     <Route exact path="/transactions" component={TransactionsPage} />
     <Route exact path="/markets/list" component={MarketListPage} />
-    <Route exact path="/markets/:id/:view?/:shareId?" component={MarketDetailPage} />
+    <Route
+      exact
+      path="/markets/:id/:view?/:shareId?"
+      render={(props) => {
+        if (web3.utils.isAddress(props.match.params.id)) {
+          return <MarketDetailPage {...props} />
+        }
+        return <Redirect to="/markets/list" />
+      }}
```

Comment:
Can you move this function out of the route definiton into a seperate variable?

Reply:
@andre-meyer Sorry, don't get it. What do you mean?

Reply:
```const marketDetailRender = (props) => {
        if (web3.utils.isAddress(props.match.params.id)) {
          return <MarketDetailPage {...props} />
        }
        return <Redirect to="/markets/list" />
      }}```